fe/fe-core/src/main/java/org/apache/doris/qe/VariableMgr.java

```diff
@@ -304,17 +286,30 @@ public static void setLowerCaseTableNames(int mode) throws DdlException {
 
     // global variable persistence
     public static void write(DataOutputStream out) throws IOException {
-        defaultSessionVariable.write(out);
+        SessionVariable variablesToWrite = defaultSessionVariable;
+        if (Catalog.isCheckpointThread()) {
```

Comment:
There should not be non-checkpoint-thread calling this method. Why not directly use defaultSessionVariableForCkpt instead.

Reply:
Not exactly, the `write` method may be called in:
1. Checkpoint thread
2. when calling `/dump` http api